### PR TITLE
Throw lock exceptions if remote share returned 423 status code

### LINKED ISSUE #17275: Put file on remote share seems to bypass lock
### Steps
1. Setup two servers A and B
1. Enable transactional file locking on A
1. Create a folder "local" on A
1. Share "local" with a user on server B
1. With cadaver/webdav on A: upload a file "test.txt" (make it slow by adding `sleep(30)` here https://github.com/owncloud/core/blob/b476c125c0e9a57a4953bea9f9f119d3ae1b1bf0/lib/private/connector/sabre/file.php#L186
1. During upload, use another cadaver/webdav instance on B and upload a file called "test.txt"

### Expected result
Upload to B fails with 423 Locked.

### Actual result
Upload to B succeeds.

Reversing the upload order works properly.
It looks like publicwebdav is somehow bypassing the locks.

### Versions
ownCloud 8.1 RC2

### PR DESCRIPTION
**Pull request by PVince81**
Fixes https://github.com/owncloud/core/issues/17275

Requires https://github.com/owncloud/core/pull/17268 to be merged first to make this testable.

### CODE REVIEW

**Comment by PVince81** [comment]
423 error is converted on upload, download and any other operations like rename on the remotely locked file.

File: lib/private/files/storage/dav.php [hunk 1]
```diff
@@ -787,6 +793,9 @@ public function hasUpdated($path, $time) {
 	private function convertException(Exception $e) {
 		Util::writeLog('files_external', $e->getMessage(), Util::ERROR);
 		if ($e instanceof ClientHttpException) {
+			if ($e->getHttpStatus() === 423) {
+				throw new \OCP\Lock\LockedException('');
```
Comment (nickvergessen) [inline lib/private/files/storage/dav.php]: Inject the path? (at least optional?)

Comment (PVince81) [inline lib/private/files/storage/dav.php]: Yeah, I was just too lazy...

**Comment by PVince81** [comment]
- [ ] TODO: rebase after https://github.com/owncloud/core/pull/17268 is merged
- [ ] TODO: inject path for exception

**Comment by PVince81** [comment]
I rebased and injected the path, except for rename/copy where we don't know which one might have caused the lock issue.

**Comment by nickvergessen** [comment]
Thanks :+1:

**Comment by PVince81** [comment]
Jenkins PR here: https://github.com/owncloud/core/pull/17296

@schiesbn @icewind1991 second review ?

**Comment by PVince81** [comment]
Rebased onto master to include the dependent fix and make it easier to review.

Please review @schiesbn @icewind1991 @MorrisJobke

**Comment by PVince81** [comment]
@owncloud-bot I hate you

Here, have a treat https://github.com/owncloud/core/pull/17363

**Comment by nickvergessen** [comment]
Jenkins passed: https://github.com/owncloud/core/pull/17363#issuecomment-118258058

**Comment by PVince81** [comment]
needs a second reviewer @schiesbn @icewind1991 @MorrisJobke

**Comment by MorrisJobke** [comment]
@PVince81 I simply can't reproduce the original issue :(

**Comment by PVince81** [comment]
@MorrisJobke maybe you didn't enable locking properly ? Double check your server A (from the steps of https://github.com/owncloud/core/issues/17275). Try uploading a file directly to A with cadaver while the file is locked. If you get 423, then try the same from server B.

Before this PR server B would tell you upload succeeded even though it didn't. After this PR server B should properly notice the locked error and forward it.

**Comment by PVince81** [comment]
@MorrisJobke can you retest ? Make sure to have master merged into this branch when you do.

**Comment by MorrisJobke** [comment]
Tested and works as expected :+1:

**Comment by MorrisJobke** [comment]
@PVince81 #17275 is targeted for 8.1.1 ... backport?

**Comment by PVince81** [comment]
Yes, it fits with locking.

@karlitschek see question above.

**Comment by karlitschek** [comment]
Yes. Backport. Thanks a lot :+1:

**Comment by MorrisJobke** [comment]
@PVince81 Can you open the backport? Thanks :)

**Comment by PVince81** [comment]
Backport PR here https://github.com/owncloud/core/pull/17710

**Pull request merged**